canvas_modules/common-canvas/src/palette/palette-dialog-content.jsx

```diff
@@ -51,7 +51,7 @@ class PaletteDialogContent extends React.Component {
 		if (categories) {
 			for (var idx = 0; idx < categories.length; idx++) {
 				if (out.indexOf(categories[idx].label) === -1) {
-					out.push(categories[idx].label);
+					out.push({ "id": categories[idx].id, "label": categories[idx].label });
 				}
```

Comment:
As per my comment above. This could just push the category object onto the `out` array.